modules/openfast-registry/src/registry_gen_fortran.cpp

```diff
@@ -133,11 +133,14 @@ void Registry::gen_fortran_module(const Module &mod, const std::string &out_dir)
 
         // If derived data type should only contain reals,
         // verify that it does, otherwise exit with error
-        if ((ddt.interface != nullptr) && ddt.interface->only_reals)
-            if (!ddt.only_contains_reals())
+        if (!this->input_int_allow)
             {
-                std::cerr << "Registry warning: Data type '" << dt_name << "' contains non-real values." << std::endl;
-                exit(EXIT_FAILURE);
+            if ((ddt.interface != nullptr) && ddt.interface->only_reals)
```

Comment:
Instead of adding a new flag, you could disable this check if the `-noextrap` flag has been set: `if ((ddt.interface != nullptr) && ddt.interface->only_reals && !this->no_extrap_interp)` and then add that flag to for this module. I didn't see extrap/interp being called for ExternalInflow.